crates/rpc-types-engine/src/sidecar.rs

```diff
@@ -51,6 +51,20 @@ impl ExecutionPayloadSidecar {
 
     /// Returns the EIP-7685 requests
     pub const fn requests(&self) -> Option<&Requests> {
-        self.prague.as_ref()
+        if let Some(RequestsOrHash::Requests(ref requests)) = self.prague {
+            Some(requests)
+        } else {
+            None
+        }
+    }
+
+    /// Calculates or retrieves the requests hash.
+    ///
+    /// - If the `prague` field contains a list of requests, it calculates the requests hash
+    ///   dynamically.
+    /// - If it contains a precomputed hash (used for testing), it returns that hash directly.
+
+    pub fn requests_hash(&self) -> B256 {
+        self.prague.as_ref().unwrap().requests_hash()
     }
```

Comment:
Please make this an `Option<B256>`, we don't want to unwrap here. In the case `self.prague` is `None`, we should just return `None`.

Reply:
> Please make this an `Option<B256>`, we don't want to unwrap here. In the case `self.prague` is `None`, we should just return `None`.

done